kernel/rtlil.cc

```diff
@@ -382,27 +382,27 @@ void RTLIL::Selection::optimize(RTLIL::Design *design)
 	std::vector<RTLIL::IdString> del_list, add_list;
 
 	del_list.clear();
-	for (auto mod_name : selected_modules) {
+	for (const auto &mod_name : selected_modules) {
```

Comment:
These and the ones below are all references to integers. How is it an improvement to keep a 64-bit pointer around that has to be dereferenced every time it's used instead of just copying the singe 32-bit integer that is being referenced?

Reply:
it is not doing that, it actually will store it locally and make not any difference in the code output (similar to https://godbolt.org/z/v-Huvy ).
Here the `const auto &` merely serves as readability improvement for the programmer when scanning the code, because it makes sure there is no _expensive_ copy.

Reply:
Only if the compiler has perfect visibility into the `count()`, `push_back()`, and `erase()` methods of the containers in question _and_ can use it to prove that there is no aliasing happening here, which may or may not be the case, but your compiler explorer code example does nothing to answer that question.

Reply:
That is true, in case of aliasing, the compiler might have to generate more expensive defensive code.

Here, we see a significant shorter code generated with the const reference; with the copy, there is our friend, the reference counter in and out of `IdString()/~IdString()`
```
make && objdump -l -C -d -S ./kernel/rtlil.o | awk '/rtlil.cc:385$/ { do_print=1; }  { if (do_print) print $0; } /rtlil.cc:390$/ { do_print=0; }' > rtlil.1
```